src/components/build/BuildInformation.test.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import BuildInformation from './BuildInformation';
+
+const setup = (propOverrides = {}) => {
+  const defaultProps = {
+    build: {},
+    appName: ''
+  };
+
+  const props = { ...defaultProps, propOverrides };
+  const wrapper = shallow(<BuildInformation {...props} />);
+
+  return { props, wrapper };
+};
+
+describe('BuildInformation', () => {
+  const {
+    props: { build },
+    wrapper
+  } = setup({
+    appName: 'myapp',
+    build: {
+      buildUrl: 'http://build.url',
+      metadata: {
+        annotations: {
+          name: 'myapp',
+          'aerogear.org/download-mobile-artifact': 'hello-world',
+          'aerogear.org/download-mobile-artifact-url': 'http://download.url',
+          'openshift.io/build-config.name': 'build-config'
+        }
+      }
+    }
+  });
+
+  it('should render', () => {
+    expect(wrapper).toHaveLength(1);
+  });
+
+  describe('BuildSummary', () => {
+    it('should render', () => {
+      const BuildSummary = wrapper.find('BuildSummary');
+
+      expect(BuildSummary).toHaveLength(1);
+      expect(BuildSummary.props().build).toEqual(build);
+    });
+  });
+
+  describe('BuildPipelineStage', () => {
+    it('should render', () => {
+      const BuildPipelineStage = wrapper.find('BuildPipelineStage');
+
+      expect(BuildPipelineStage).toHaveLength(1);
+      expect(BuildPipelineStage.props().build).toEqual(build);
+    });
+  });
+
+  describe('BuildDownload', () => {
+    it('should match previous snapshot', () => {
+      const BuildDownload = wrapper.find('BuildDownload');
+
+      expect(BuildDownload).toMatchSnapshot();
```

Comment:
Since `BuildDownload` is a container component which is connected to the store, I was not sure of the bet way to test it from here, since it does not act like regular components. 

This compares the JSON output of the component to the previous time and fails if they do not match.

If you think this is a bad approach please let me know and I will look at changing it.

Reply:
The pattern we have used in MDC to test container component is:

1. Export both components (the one that is connected to the store, and the one that isn't) from the container class.
2. In the test, only test the component that is not connected to the store.

A good example is this: https://github.com/aerogear/mobile-developer-console/blob/master/src/containers/Client.test.js.

This way you can still test the component without building a store.

Reply:
Thanks.